rpi-eeprom-update

```diff
@@ -237,7 +241,25 @@ applyRecoveryUpdate()
 
    [ "${BOOTLOADER_CURRENT_VERSION}" -ge "${RPI_EEPROM_SELF_UPDATE_MIN_VER}" ] || RPI_EEPROM_SELF_UPDATE=0
 
-   if [ "${RPI_EEPROM_SELF_UPDATE}" != "1" ]; then
+   # For immediate updates via flash the recovery.bin update is created and then discarded if the
+   # flashrom update was successful. For SD boot (most common) this provides a rollback in the event
+   # of power loss.
+   if [ "${RPI_EEPROM_USE_FLASHROM}" = 1 ]; then
+      echo
+      echo "UPDATING bootloader. Do not disconnect the power until the update is complete"
+      echo
+      FLASHROM_LOG="$(mktemp)"
+      if flashrom -p linux_spi:dev=${SPIDEV},spispeed=16000 -w "${BOOTFS}/pieeprom.upd" > "${FLASHROM_LOG}"; then
+         # Success - remove fallback fails
+         removePreviousUpdates
+         echo "UPDATE SUCCESSFUL"
+      else
+         # Leave the recovery files in case the EEPROM has been partially updated
+         cat "${FLASHROM_LOG}"
+         die "UPDATE FAILED"
+      fi
+      return
+   elif [ "${RPI_EEPROM_SELF_UPDATE}" != "1" ]; then
```

Comment:
Won't "${RPI_EEPROM_SELF_UPDATE}" != "1" always be true?

Reply:
It defaults to zero but could be forced to 1 via the /etc/defaults file

Reply:
I'm guessing the point is more that this line could be replaced with `else`.

Reply:
^

Reply:
No, either it's changed or (more likely) I misread it - the `if` is about `RPI_EEPROM_USE_FLASHROM`.